src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/ContigAlignmentsModifier.java

```diff
@@ -0,0 +1,208 @@
+import org.broadinstitute.hellbender.tools.spark.sv.discovery.AlignedContig;
+import org.broadinstitute.hellbender.tools.spark.sv.discovery.AlignmentInterval;
+import org.broadinstitute.hellbender.tools.spark.sv.utils.SvCigarUtils;
+import org.broadinstitute.hellbender.utils.IntervalUtils;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.Utils;
+import scala.Tuple2;
+import scala.Tuple3;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+public final class ContigAlignmentsModifier {
+
+    /**
+     * Removes overlap between input {@code contig}'s two alignments.
+     * If the two alignment intervals are NOT overlapping, return the original aligned contig.
+     */
+    public static List<AlignmentInterval> removeOverlap(final AlignmentInterval one, final AlignmentInterval two,
+                                                        final SAMSequenceDictionary dictionary) {
+
+        final AlignmentInterval reconstructedOne, reconstructedTwo;
+        final int overlapOnRead = AlignmentInterval.overlapOnContig(one, two);
+        if (overlapOnRead == 0) {
+            reconstructedOne = one;
+            reconstructedTwo = two;
+        } else {
+            final boolean oneYieldToTwo;
+            if (one.referenceSpan.getContig().equals(two.referenceSpan.getContig())) {
+                if (one.forwardStrand != two.forwardStrand) { // so that the inverted duplicated reference span is minimal.
+                    // jumpStart is for "the starting reference location of a jump that linked two alignment intervals", and
+                    // jumpLandingRefLoc is for "that jump's landing reference location"
+                    final int jumpStartRefLoc = one.referenceSpan.getEnd(),
+                            jumpLandingRefLoc = two.referenceSpan.getStart();
+                    oneYieldToTwo = jumpStartRefLoc <= jumpLandingRefLoc == one.forwardStrand;
+                } else {
+                    oneYieldToTwo = one.forwardStrand;
+                }
+            } else {
+                oneYieldToTwo = IntervalUtils.compareContigs(one.referenceSpan, two.referenceSpan, dictionary) > 0;
+            }
+
+            if (oneYieldToTwo) {
+                reconstructedOne = clipAlignmentInterval(one, overlapOnRead, true);
+                reconstructedTwo = two;
+            } else {
+                reconstructedOne = one;
+                reconstructedTwo = clipAlignmentInterval(two, overlapOnRead, false);
+            }
+
+        }
+        return Arrays.asList(reconstructedOne, reconstructedTwo);
+    }
+
+    /**
+     * Given {@code clipLengthOnRead} to be clipped from an aligned contig's {@link AlignmentInterval} {@code input},
+     * return a modified {@link AlignmentInterval} with the requested number of bases clipped away and ref span recomputed.
```

Comment:
I'd document here that the NM field will be wiped out.

Reply:
done.